src/cargo/core/features.rs

```diff
@@ -655,6 +706,7 @@ impl CliUnstable {
         fn parse_features(value: Option<&str>) -> Vec<String> {
             match value {
                 None => Vec::new(),
+                Some("") => Vec::new(),
```

Comment:
Is there a particular reason to add this?

Reply:
Yes — it makes it so that `-Zallow-features=` is parsed as a list of _no_ features, not as a list holding a single feature whose name is empty. The latter interpretation would still lead to the correct _result_, but would make errors weirder since they would include an empty element.

Reply:
The call to `split` should yield to an empty vector just the same.  That is:

```rust
    assert_eq!(Vec::<String>::new(),
               "".split(',').map(|s| s.to_string()).collect::<Vec<_>>());
```

Reply:
```
   Compiling playground v0.0.1 (/playground)
    Finished dev [unoptimized + debuginfo] target(s) in 5.70s
     Running `target/debug/playground`
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `[]`,
 right: `[""]`', src/main.rs:2:5
 ```

Reply:
Oh. 😆  I thought I clicked the run button.  I coulda sworn `split` would return an empty sequence.  Apologies! 😄

Reply:
Hehe, all good. I only know this because I've hand-written far too much parsing code :p